maker/src/health.rs

```diff
@@ -0,0 +1,120 @@
+// TODO: Add health collection metrics
+// This can be later used to add health metrics to the health endpoint
+
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use reqwest::Client;
+use reqwest::StatusCode;
+use reqwest::Url;
+use std::time::Duration;
+use tokio::sync::watch;
+
+/// Health status of a service
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum ServiceStatus {
+    #[default]
+    Unknown,
+    Online,
+    Offline,
+}
+
+/// Health monitoring for the node
+///
+/// Simple endpoint querying is handled by provided configuration, for more complex health checks
+/// the transmitters are exposed to be plugged in the services that need to publish their health
+/// status.
+
+#[derive(Clone)]
+pub struct Health {
+    /// Coordinator HTTP API status
+    coordinator_rx: watch::Receiver<ServiceStatus>,
+    /// Orderbook websocket stream status
+    orderbook_rx: watch::Receiver<ServiceStatus>,
+}
+
+/// Transmitters that need to be plugged in the services that need to publish their health status.
+pub struct Tx {
+    pub orderbook: watch::Sender<ServiceStatus>,
+    pub coordinator: watch::Sender<ServiceStatus>,
+}
+
+impl Health {
+    pub fn new() -> (Self, Tx) {
+        let (orderbook_tx, orderbook_rx) = watch::channel(ServiceStatus::Unknown);
+        let (coordinator_tx, coordinator_rx) = watch::channel(ServiceStatus::Unknown);
+
+        (
+            Self {
+                coordinator_rx,
+                orderbook_rx,
+            },
+            Tx {
+                orderbook: orderbook_tx,
+                coordinator: coordinator_tx,
+            },
+        )
+    }
+
+    // TODO: Any ideas of the most useful way of sending this information?
+    // - perhaps a serialized struct (json) with the status of each service?
+    pub fn get_health(&self) -> Result<String> {
+        let mut health_info = String::new();
+        health_info.push_str(&format!(
+            "Coordinator: {:?}\n",
+            self.get_coordinator_status()
+        ));
+        health_info.push_str(&format!("Orderbook: {:?}\n", self.get_orderbook_status()));
+
+        if self.get_coordinator_status() == ServiceStatus::Online
+        // TODO: Uncomment this line when we retrieve the orderbook status
+        // && (self.get_orderbook_status() == ServiceStatus::Online)
```

Comment:
Does this work already?

Reply:
if by "orderbook" we mean websocket stream that other people look at orders, then not, we are still not subscribing to it from the maker. I was hoping that other work done for this milestone would encompass this, but nothing got still merged.

@luckysori do I remember correctly that you were adding the orderbook websocket stream?

Reply:
edit: I have removed the TODO from this place and hard-coded `Online` status for the orderbook where it should be plugged in (there's another TODO there anyway).